docs/enhancements/infrastructure-monitoring-service-api/alarms.md

```diff
@@ -0,0 +1,797 @@
+        AlarmAcknowledgedTime    // Done through API request AlarmEventRecordModifications.alarmAcknowledged (update AlarmAcknowledged)
+}
+```
+
+## Schema
+All O-RAN services will use the same O-CLOUD DB service. More on DB deployment [here](#postgres).
+
+Each table is modeled after O-RAN data structures. DB in our case wil be called `alarms`.
+Init SQL may look like the following:
+```sql
+CREATE DATABASE alarms;
+
+-- ENUM for ManagementInterfaceID
+DROP TYPE IF EXISTS ManagementInterfaceID CASCADE;
+CREATE TYPE ManagementInterfaceID AS ENUM ('O1', 'O2DMS', 'O2IMS', 'OpenFH');
+
+-- ENUM for AlarmLastChangeType
+DROP TYPE IF EXISTS AlarmLastChangeType CASCADE;
+CREATE TYPE AlarmLastChangeType AS ENUM ('added', 'deleted', 'modified');
+
+DROP TYPE IF EXISTS ClearingType CASCADE;
+CREATE TYPE ClearingType AS ENUM ('automatic', 'manual');
+
+-- Table: versions, link alarm_dictionary and alarm_definitions using version.
+DROP TABLE IF EXISTS versions CASCADE;
+CREATE TABLE versions (
+         version_number VARCHAR(50) PRIMARY KEY
+);
+
+-- Table: alarm_dictionary
+DROP TABLE IF EXISTS alarm_dictionary CASCADE;
+CREATE TABLE alarm_dictionary (
+          resource_type_id UUID PRIMARY KEY,
+          alarm_dictionary_version VARCHAR(50) NOT NULL,  -- Links alarm_dictionary and alarm_definitions
+          alarm_dictionary_schema_version VARCHAR(50) DEFAULT 'TBD-O-RAN-DEFINED' NOT NULL,
+          entity_type VARCHAR(255) NOT NULL,
+          vendor VARCHAR(255) NOT NULL,
+          management_interface_id ManagementInterfaceID[] DEFAULT ARRAY['O2IMS']::ManagementInterfaceID[],
+          pk_notification_field TEXT[] DEFAULT ARRAY['resource_type_id']::TEXT[],
+          created_at TIMESTAMPTZ DEFAULT CURRENT_TIMESTAMP,
+          CONSTRAINT fk_alarm_dictionary_version FOREIGN KEY (alarm_dictionary_version) REFERENCES versions(version_number)
+);
+
+-- Table: alarm_definitions
+DROP TABLE IF EXISTS alarm_definitions CASCADE;
+CREATE TABLE alarm_definitions (
+           alarm_definition_id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+           alarm_name VARCHAR(255) NOT NULL,
+           alarm_last_change VARCHAR(50) NOT NULL,
+           alarm_description TEXT NOT NULL,
+           proposed_repair_actions TEXT NOT NULL,
+           alarm_dictionary_version VARCHAR(50) NOT NULL, -- Links alarm_dictionary and alarm_definitions
+           alarm_additional_fields JSONB,
+           alarm_change_type AlarmLastChangeType DEFAULT 'added' NOT NULL,
+           clearing_type ClearingType DEFAULT 'automatic' NOT NULL,
+           management_interface_id ManagementInterfaceID[] DEFAULT ARRAY['O2IMS']::ManagementInterfaceID[],
+           pk_notification_field TEXT[] DEFAULT ARRAY['alarm_definition_id']::TEXT[],
+           created_at TIMESTAMPTZ DEFAULT CURRENT_TIMESTAMP,
+           CONSTRAINT fk_alarm_definitions_version FOREIGN KEY (alarm_dictionary_version) REFERENCES versions(version_number),
```

Comment:
This method of linking these two tables seems a bit more complicated than necessary.  I would prefer to see a column in this table for `alarm_dictionary_id` and a FK constraint pointing directly to the other table since these are 1:N rather than M:N.